File: src/main/kotlin/cn/staynoob/social/provider/qq/autoconfigure/QQProperties.java
package cn.staynoob.social.provider.qq.autoconfigure;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.NotNull;

@Validated
@ConfigurationProperties(prefix = "social.qq")
public class QQProperties {

    @NotNull
    private String clientId;
    @NotNull
    private String clientSecret;

    public String getClientId() {
        return clientId;
    }

    public void setClientId(String clientId) {
        this.clientId = clientId;
    }

    public String getClientSecret() {
        return clientSecret;
    }

    public void setClientSecret(String clientSecret) {
        this.clientSecret = clientSecret;
    }

}

File: src/main/kotlin/cn/staynoob/social/provider/qq/autoconfigure/QQProperties.kt
package cn.staynoob.social.provider.qq.autoconfigure

import cn.staynoob.social.kotlin.KotlinAllOpen
import cn.staynoob.social.kotlin.NoArgConstructor
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated
import javax.validation.constraints.NotNull

@Validated
@KotlinAllOpen
@NoArgConstructor
@ConfigurationProperties(prefix = "social.qq")
class QQProperties(
        @get: NotNull
        var appId: String? = null,
        @get: NotNull
        var appSecret: String? = null
)